### PR TITLE
Updated Mithril defs to match 1.0.0.

### PR DESCRIPTION
**Pull request by andraaspar**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://mithril.js.org/change-log.html
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: mithril/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Type definitions for Mithril
```
Comment (tlaziuk) [inline mithril/index.d.ts]: you should add version here, like 1.0

**Comment by andraaspar** [comment]
I've addressed the request, plus added some improvements. Is this OK to merge?

**Comment by zhengbli** [comment]
@tlaziuk further comments?

**Comment by tlaziuk** [comment]
@zhengbli I think this typings is no longer needed, mithril probably would release typings along with their main package by version 1.0.2, as the typings file appeared after 1.0.1 release -- https://github.com/lhorie/mithril.js/blob/next/mithril.d.ts

**Pull request closed without merging**